File: app/src/main/java/com/josh/billrssroom/ui/favorites/FavoritesAdapter.java
package com.josh.billrssroom.ui.favorites;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.josh.billrssroom.R;
import com.josh.billrssroom.databinding.ItemRowFavoritesBinding;
import com.josh.billrssroom.model.BillItem;

import java.util.List;

public class FavoritesAdapter extends RecyclerView.Adapter<FavoritesViewHolder> {

    private final FavoriteClickListener favoriteClickListener;
    private List<BillItem> billItems;


    public FavoritesAdapter(Context context, FavoriteClickListener favoriteClickListener) {
        this.favoriteClickListener = favoriteClickListener;
    }

    @NonNull
    @Override
    public FavoritesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        ItemRowFavoritesBinding binding = DataBindingUtil
                .inflate(LayoutInflater.from(parent.getContext()), R.layout.item_row_favorites,
                        parent, false);
        return new FavoritesViewHolder(binding, favoriteClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull FavoritesViewHolder holder, int position) {
        BillItem item = billItems.get(position);
        holder.binding.setBill(billItems.get(position));
        holder.binding.executePendingBindings();

        holder.billItem = item;
    }

    public void setFavoriteItems(List<BillItem> billEntities) {
        this.billItems = billEntities;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return billItems == null ? 0 : billItems.size();
    }
}

File: app/src/main/java/com/josh/billrssroom/ui/favorites/FavoritesAdapter.kt
package com.josh.billrssroom.ui.favorites

import android.app.Activity
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.josh.billrssroom.R
import com.josh.billrssroom.model.FeedItem
import kotlinx.android.synthetic.main.item_row_favorites.view.*
import java.util.*

class FavoritesAdapter(private val activity: Activity,
                       val favoriteClickCallback: FavoriteClickListener) :
    RecyclerView.Adapter<FavoritesAdapter.FavoritesViewHolder>() {

    private var favoritesList: List<FeedItem> = Collections.emptyList()

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): FavoritesViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        return FavoritesViewHolder(layoutInflater.inflate(R.layout.item_row_favorites, parent, false))
    }

    override fun onBindViewHolder(holder: FavoritesViewHolder, position: Int) {
        val items = favoritesList.get(position)
        // or? val billModel = favoritesList[position]
        // or? holder.bindTo(favoritesList[position])

        holder.bindTo(items)
    }

    fun setBillItemList(billFeedItemList: List<FeedItem>) {
        favoritesList = billFeedItemList
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return favoritesList.size
    }

    inner class FavoritesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val titleView = itemView.text_title
        val dateView = itemView.text_pubdate
        val descriptionView = itemView.text_description
        val btnShare = itemView.btn_share
        val btnBrowser = itemView.btn_browser
        val btnTrash = itemView.btn_trash
        var favorite: FeedItem? = null

        fun bindTo(favorite: FeedItem){
            this.favorite = favorite
            titleView.text = favorite.title
            dateView.text = favorite.formattedDate
            descriptionView.text = favorite.formattedDescription

            btnTrash.setOnClickListener { favoriteClickCallback.onTrashClick(favorite, adapterPosition) }
            btnBrowser.setOnClickListener { favoriteClickCallback.onBrowserClick(favorite, adapterPosition) }
            btnShare.setOnClickListener { favoriteClickCallback.onShareClick(favorite, adapterPosition) }
        }
    }
}